scripts/start.js

```diff
@@ -0,0 +1,87 @@
+const { spawn } = require('child_process');
+const webpack = require('webpack');
+const WebpackDevServer = require('webpack-dev-server');
+const chalk = require('chalk');
+const waitOn = require('wait-on');
+
+const rendererConfig = require('../webpack.renderer.js');
+
+const PORT = process.env.PORT || 8601;
+
+const buildRenderer = () => {
+    return new Promise((resolve, reject) => {
+        console.log(chalk.cyan('Building renderer process...'));
+
+        const compiler = webpack(rendererConfig);
+        compiler.run((err, stats) => {
+            if (err || stats.hasErrors()) {
+                console.error(chalk.red('Renderer build failed:', err || stats.toString()));
+                reject(err || new Error('Renderer build failed.'));
+            } else {
+                console.log(chalk.green('Renderer built successfully!'));
+                resolve();
+            }
+        });
+    });
+};
+
+const startRenderer = async () => {
+    console.log(chalk.cyan('Starting Webpack Dev Server...'));
+
+    const compiler = webpack(rendererConfig);
+    const server = new WebpackDevServer(compiler, {
+        hot: true,
+        compress: true,
+        port: PORT,
+        headers: { 'Access-Control-Allow-Origin': '*' },
+        historyApiFallback: true
+    });
+
+    return new Promise((resolve, reject) => {
+        server.listen(PORT, 'localhost', (err) => {
+            if (err) {
+                console.error(chalk.red('Failed to start Webpack Dev Server:', err));
+                reject(err);
+            } else {
+                console.log(chalk.green(`Renderer is running at http://localhost:${PORT}`));
+                resolve();
+            }
+        });
+    });
+};
+
+
+const startElectron = async () => {
+    console.log(chalk.cyan('Starting Electron...'));
+
+    await waitOn({ resources: [`http://localhost:${PORT}`] });
+
+    spawn('electron', ['.'], {
+        stdio: 'inherit',
+        shell: true
+    });
+};
+
+const start = async () => {
+    console.log(chalk.green('Building main process...'));
+
+    const mainProcess = spawn('npm', ['run', 'compile:main'], {
+        stdio: 'inherit',
+        shell: true
+    });
+
+    mainProcess.on('exit', async (code) => {
```

Comment:
Hmm... can't you do something like `await spawn(...)`?
_(does some web searching)_
...nevermind, this looks good. 😅